runtime/runtime-params-estimator/src/lib.rs

```diff
@@ -441,77 +468,124 @@ fn action_stake(ctx: &mut EstimatorContext) -> GasCost {
     total_cost - base_cost
 }
 
-/// For deploy costs, the cost is the sum two components:
-///   * database-related costs which we measure by deploying a contract with
-///     dummy payload
-///   * cost of contract compilation, which we estimate using linear regression
-///     on some real contracts.
 fn action_deploy_contract_base(ctx: &mut EstimatorContext) -> GasCost {
-    let (compilation_base_cost, _) = compilation_cost_base_per_byte(ctx);
-    let base_cost = deploy_contract_base(ctx);
-    base_cost + compilation_base_cost
-}
-fn action_deploy_contract_per_byte(ctx: &mut EstimatorContext) -> GasCost {
-    let total_cost = {
-        let code = read_resource(if cfg!(feature = "nightly_protocol_features") {
-            "test-contract/res/nightly_large_contract.wasm"
-        } else {
-            "test-contract/res/stable_large_contract.wasm"
-        });
-        deploy_contract_cost(ctx, code)
-    };
-
-    let (_, compilation_per_byte_cost) = compilation_cost_base_per_byte(ctx);
-    let base_cost = deploy_contract_base(ctx);
-
-    let bytes_per_transaction = 1024 * 1024;
-
-    (total_cost - base_cost) / bytes_per_transaction + compilation_per_byte_cost
-}
-fn deploy_contract_base(ctx: &mut EstimatorContext) -> GasCost {
     if let Some(cost) = ctx.cached.deploy_contract_base.clone() {
         return cost;
     }
 
-    let total_cost = {
+    let cost = {
         let code = read_resource("test-contract/res/smallest_contract.wasm");
-        deploy_contract_cost(ctx, code)
+        deploy_contract_cost(ctx, code, Some(b"sum"))
     };
 
-    let base_cost = action_sir_receipt_creation(ctx);
-
-    let cost = total_cost - base_cost;
     ctx.cached.deploy_contract_base = Some(cost.clone());
     cost
 }
-fn deploy_contract_cost(ctx: &mut EstimatorContext, code: Vec<u8>) -> GasCost {
+fn action_deploy_contract_per_byte(ctx: &mut EstimatorContext) -> GasCost {
+    let mut xs = vec![];
+    let mut ys = vec![];
+
+    for (contract, pivot_fn) in REAL_CONTRACTS_SAMPLE {
+        let code = read_resource(contract);
+        xs.push(code.len() as u64);
+        let cost = deploy_contract_cost(ctx, code, Some(pivot_fn.as_bytes()));
+        ys.push(cost);
+    }
```

Comment:
There's a bit of a problem here. Previous estimate has this bit of logic in it: 

```
 // We multiply `b` by 5/4 to accommodate for the fact that test contracts are typically 80% code,
    // so in the worst case it could grow to 100% and our costs still give better upper estimation.
    // Safety muliplication with 5/4.
    let safety_numer = 5u64;
    let safety_denom = 4u64;
```

I think we should preserve it. 

Basically, any contract contains some data, and some code. It takes almost no time to compile data, but compiling code is slow. The sample of *real* contracts we use contains about 20% of data, and we should account for that.